src/internal/converters/ics/ics.go

```diff
@@ -175,10 +174,19 @@ func getRecurrencePattern(
 		case models.ENDDATE_RECURRENCERANGETYPE:
 			end := rrange.GetEndDate()
 			if end != nil {
-				// NOTE: We convert just a date into date+time in a
-				// different timezone which will cause it to not be just
-				// a date anymore.
-				endTime, err := getUTCTime(end.String(), ptr.Val(rrange.GetRecurrenceTimeZone()))
+				parsedTime, err := dttm.ParseTime(end.String())
```

Comment:
IIUC parsedTime the start time of the end date? Which is why we need to add 23 hours 59 minutes 59 seconds in the lines below?

Reply:
This is recurrence end date, which should be treated as the last minute of the day so that any events on that day is also counted.